### ISSUE TITLE
Add libxml2 to hunter-packages

### ISSUE THREAD

**Issue by richardeigenmann** [ruslo/hunter#1174]
As suggested by @headupinclouds  in  #1173 I am requesting that libxml2 be added to hunter-packages.
I notice that the existing hunter/cmake/projects/libxml2/hunter.cmake pulls down version "http://xmlsoft.org/sources/libxml2-2.9.4.tar.gz" whilst 2.9.7 is available.

Afterwards David suggests that I would then add a CMakeLists.txt like the following to the repo.

Yet I fear we then still need to figure out how to get iconv into hunter which libxml2 is dependent on.
https://ftp.gnu.org/pub/gnu/libiconv/

cmake_minimum_required(VERSION 2.8.10)
project( xml2 )

if(MINGW OR UNIX)
  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-attributes")
endif()

set(XML2_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
include_directories( "${XML2_SOURCE_DIR}/include" "${CMAKE_INSTALL_PREFIX}/include")
LINK_DIRECTORIES("${CMAKE_INSTALL_PREFIX}/lib")

#for windows only must fix later
include_directories("${XML2_SOURCE_DIR}/win32/VC10")

file( GLOB XML2_HEADERS ${XML2_SOURCE_DIR}/*.h )
file( GLOB XML2_SOURCES ${XML2_SOURCE_DIR}/*.c )

file( GLOB API_HEADERS ${XML2_SOURCE_DIR}/include/libxml/*.h )

#source_group( libxml FILES ${GLEW_HEADERS} )
#source_group( src    FILES ${GLEW_SOURCES} )
#include_directories(${ANDROID_NDK}/platforms/android-14/arch-arm/usr/include/asm)
#MESSAGE(${ANDROID_NDK}/platforms/android-14/arch-arm/usr/include)

if(XML2_BUILD_SHARED)
	set(XML2_LIBTYPE SHARED)
else()
	set(XML2_LIBTYPE STATIC)
	add_definitions( "-DLIBXML_STATIC" )
endif()

add_library( xml2 ${XML2_LIBTYPE} ${XML2_SOURCES} ${XML2_HEADERS} )
target_link_libraries(xml2 iconv ws2_32)

set_target_properties(xml2 PROPERTIES PUBLIC_HEADER "${API_HEADERS}")

set(BINDIR bin)
set(BINCOMPONENT runtime)
set(LIBDIR lib)
set(LIBCOMPONENT dev)
set(INCLUDEDIR include/libxml)
set(INCLUDECOMPONENT dev)

install(TARGETS xml2
	RUNTIME DESTINATION ${BINDIR} COMPONENT ${BINCOMPONENT}
	LIBRARY DESTINATION ${LIBDIR} COMPONENT ${LIBCOMPONENT}
	ARCHIVE DESTINATION ${LIBDIR} COMPONENT ${LIBCOMPONENT}
	PUBLIC_HEADER DESTINATION ${INCLUDEDIR} COMPONENT ${INCLUDECOMPONENT})

**Comment by ruslo** [comment]
Created https://github.com/hunter-packages/libxml2

@richardeigenmann I've added you to the corresponding team. After you'll accept invitation (check email) you will have write access. Please read:
* https://docs.hunter.sh/en/latest/creating-new/patch.html

> whilst 2.9.7 is available

Branch `hunter-2.9.7` created:
* https://github.com/hunter-packages/libxml2/tree/hunter-2.9.7

**Comment by richardeigenmann** [comment]
Thanks but I am new here and confused.

Do I understand that you created a branch for libxml 2.9.7 in the https://github.com/hunter-packages/libxml2 repo?

Would I now try to add the CMakeLists.txt to that branch? I would then use my Win10 laptop to run cmake and pull down libxml2 from the new branch with a hunter_add_package( libxml2 ) command? Eventually I would figure out what it takes to make Visual Studio 2017 like libxml2 and compile it and commit that CMakeLists.txt file?

How do I tell Hunter which branch to pull the lixbml2 from? 
Would I not be mucking up the repo in my own GitHub space before making a pull request? How would I tell Hunter to search in my space?

**Comment by headupinclouds** [comment]
> Do I understand that you created a branch for libxml 2.9.7 in the https://github.com/hunter-packages/libxml2 repo?

Yes.  See https://github.com/hunter-packages/libxml2/tree/hunter-2.9.7

> Would I now try to add the CMakeLists.txt to that branch? 

Assuming that path looks feasible, yes.   As @xsacha mentioned in [this post](https://github.com/ruslo/hunter/issues/1173#issuecomment-343363285), it can be non trivial to convert some autoconf projects to CMake.  You can also use a Hunter `autoconf` scheme at the expensive of some portability (no MSVC).  Since there was already a CMakeLists.txt file for this one, it seems worth testing.

> I would then use my Win10 laptop to run cmake and pull down libxml2 from the new branch with a hunter_add_package( libxml2 ) command? 

Be sure to read through https://docs.hunter.sh/en/latest/creating-new.html

There is also a recent related discussion here: https://github.com/ruslo/hunter/issues/1125

Here is a brief summary:

* add your changes to https://github.com/hunter-packages/libxml2/tree/hunter-2.9.7
* draft a new release of your branch called v2.9.7-p0 here https://github.com/hunter-packages/libxml2/releases
* download `wget https://github.com/hunter-packages/libxml2/archive/v2.9.7-p0.tar.gz`
* get the SHA1 `shasum v2.9.7-p0.tar.gz `
* create a fork of hunter https://github.com/ruslo/hunter#fork-destination-box
* clone your hunter fork locally `git clone https://github.com/richardeigenmann/hunter.git`
* add a package specification (see below) in hunter/cmake/projects/libxml2/hunter.cmake [example](https://github.com/ruslo/hunter/blob/master/cmake/projects/yaml-cpp/hunter.cmake)
* add a default entry in [hunter/cmake/configs/default.cmake](https://github.com/ruslo/hunter/blob/master/cmake/configs/default.cmake)  w/ format `hunter_config(libxml2 VERSION 2.9.7-p0)`
* add a build test in hunter/examples/libxml2 [example](https://github.com/ruslo/hunter/tree/master/examples/Jpeg)
* test the build from the top hunter directory: `TOOLCHAIN=default PROJECT_DIR=examples/libxml2 ./jenkins.py`
* commit your changes to something like `pr.libxml2`; push upstream; submit a PR to hunter

```
hunter_add_version(
  PACKAGE_NAME
  libxml2
  VERSION
  2.9.7-p0
  URL
  "https://github.com/hunter-packages/libxml2/archive/v2.9.7.tar.gz"
  SHA1
  9ffd9032c5fa53c7107cc22307f70fde1e088e51 # actual value TBD
  )
```

Once that change is merged you can use it from CMake in your application or library with something like:

```
hunter_add_package(libxml2)
find_package(libxml2 CONFIG REQUIRED)
target_link_libraries(yourapp libxml2::libxml2)
```

> How do I tell Hunter which branch to pull the lixbml2 from

Hunter will typically work from a tagged release.  See the `URL` entry in the sample `hunter_add_version()` block shown above.

**Comment by xsacha** [comment]
Did anyone do some work to get a CMake version compiling?
Otherwise, I have made a MSVC scheme.
